ui/src/shared/components/ScatterContainer.tsx

```diff
@@ -75,15 +86,17 @@ const ScatterContainer: FunctionComponent<Props> = ({
         onSetFillColumns(filteredGroupKeys)
       }
     }
+    if (!xColumn) {
+      onSetXColumn(chooseXColumn(table))
+    }
+    if (!yColumn) {
+      onSetYColumn(chooseYColumn(table))
+    }
   })
 
   const fillColumns = storedFill || []
   const symbolColumns = storedSymbol || []
 
-  // TODO: allow xcolumn and ycolumn to be user selectable
-  const xColumn = chooseXColumn(table)
```

Comment:
I think we need to add in some guards to make sure the `xColumn` selection is valid. Something like this:

https://github.com/influxdata/influxdb/blob/e4772f06b238065e1bc25bc642042da73e926c4b/ui/src/shared/components/HeatmapContainer.tsx#L62-L67